src/script/ui/state/rdkitjs/index.js

```diff
@@ -90,15 +91,44 @@ function deleteAllSGroupsWithName(restruct, action, fieldName) {
 	});
 }
 
-function doTransformation(method, mol) {
+/**
+ * @param method {string}
+ * @param mol {*} TODO (pradeep): Figure out a typedef for mol
+ * @param molDetails {StructDetails}
+ * @return {*}
+ */
+function doTransformation(method, mol, molDetails) {
 	switch (method) {
 		case 'aromatize':
 			return mol.get_aromatic_form();
 		case 'dearomatize':
 			return mol.get_kekule_form();
 		case 'clean':
 			return mol.get_new_coords();
+		case 'toggleExplicitHydrogen':
+			if (molDetails.hasExplicitHydrogen) {
+				return mol.remove_hs();
+			} else {
+				return mol.add_hs();
+			}
 		default:
 			return '';
 	}
 }
+
+/**
+ * @param struct {Struct}
+ * @return {StructDetails}
+ */
+function getStructDetails(struct) {
```

Comment:
Helper function that stores more details about the structure, For now, it only has 1 field.